test/gpu/native/foreachexpr.chpl

```diff
@@ -0,0 +1,35 @@
+use GpuDiagnostics;
```

Comment:
Could you add a non-gpu test so that we test this feature in other configs, too? Maybe in `test/expressions/loop-expr`? Maybe copying something that exists for `for`/`forall` and using `foreach` instead if that makes sense.

Reply:
I've added some more tests and adjusted other tests to reference foreach expressions.